config/api/src/main/java/org/jboss/windup/config/RuleSubset.java

```diff
@@ -306,10 +306,14 @@ public void perform(Rewrite rewrite, EvaluationContext context)
                     if (location != null)
                         exMsg += "\n  Defined in: " + location;
                 }
-                throw new WindupException(exMsg, ex);
 
+                Object fatal_ = ruleContext.get(RuleMetadataType.TREAT_EXCEPTIONS_AS_FATAL);
+                if (fatal_ instanceof Boolean && ((Boolean)fatal_).booleanValue())
+                    throw new WindupException(exMsg, ex);
+                // If the exception is not fatal, some RuleLifecycleListener#afterRuleExecutionFailed() will take care.
             }
-        }
+
+        }// for each Rule
```

Comment:
Well, this is really far from the opening { and I got lost in those closing braces. This could be an example where it gives you some orientation point...